dev/__test__/dependencies.test.js

```diff
@@ -1,23 +1,96 @@
+          `Use \`make dep-intersect\` at root folder to check the intersection versions among all ${targetDepType}.`
         )
       }
       expect(validatedVersion).not.toBeNull()
     })
   })
 })
+
+describe('All dev dependencies should have valid intersection of its version ranges across different packages.', () => {
+  const targetDepType = dependencyTypeShorthands.dev
+  const dependencyVersionsByDependency = utils.listDependencyVersionsByDependency(
+    targetDepType
+  )
+  _.forEach(dependencyVersionsByDependency, (ranges, dependencyName) => {
+    test(`\`${dependencyName}\` should have valid intersection`, () => {
+      const validatedVersion = utils.getIntersection(ranges)
+      if (validatedVersion === null) {
+        console.warn(
+          `Use \`make dep-intersect\` at root folder to check the intersection versions among all ${targetDepType}.`
+        )
+      }
+      expect(validatedVersion).not.toBeNull()
+    })
+  })
+})
+
+describe(`Check versions of ${dependencyTypeShorthands.dev}`, () => {
+  const devDependencyVersionsByPackage = utils.listDependencyVersionsByPackage(
+    dependencyTypeShorthands.dev
+  )
+  describe(`The version of each optional dependency should be the same as it in ${dependencyTypeShorthands.dev}`, () => {
+    const optionalDependencyVersionsByPackage = utils.listDependencyVersionsByPackage(
+      dependencyTypeShorthands.optional
+    )
+    _.forEach(
+      optionalDependencyVersionsByPackage,
+      (optionalDependencies, packageDirname) => {
+        _.forEach(optionalDependencies, (rangeOfOptional, dependencyName) => {
+          const rangeOfDev = _.get(devDependencyVersionsByPackage, [
+            packageDirname,
+            dependencyName,
+          ])
+          if (rangeOfDev) {
+            test(`The version of optional dependency \`${dependencyName}\` should be the same as it of ${dependencyTypeShorthands.dev} in package \`${packageDirname}\``, () => {
+              expect(rangeOfOptional).toBe(rangeOfDev)
+            })
+          }
+        })
+      }
+    )
+  })
+
+  describe(`The version of each peer dependency should be the same as it in ${dependencyTypeShorthands.dev}`, () => {
+    const peerDependencyVersionsByPackage = utils.listDependencyVersionsByPackage(
+      dependencyTypeShorthands.peer
+    )
+    _.forEach(
+      peerDependencyVersionsByPackage,
+      (peerDependencies, packageDirname) => {
+        _.forEach(peerDependencies, (rangeOfPeer, dependencyName) => {
+          const rangeOfDev = _.get(devDependencyVersionsByPackage, [
+            packageDirname,
+            dependencyName,
+          ])
+          if (rangeOfDev) {
+            test(`The version of peer dependency \`${dependencyName}\` should be the same as it of ${dependencyTypeShorthands.dev} in package \`${packageDirname}\``, () => {
+              expect(rangeOfPeer).toBe(rangeOfDev)
+            })
+          }
+        })
+      }
+    )
+  })
```

Comment:
not sure why `pacage.json#optionalDependencies` and `package.json#peerDependencies` should be the same as `pacakge.json#devDepenedencies` ?

Reply:
The original target scenario is that sometimes we need to install a package belongs to peerDependencies or optionalDependencies for testing and linting on the local device, so we will add it to devDepenedencies.

This test is to make sure that the version are keep the same across the dependency categories.

But after latest update I have remove all optional and peer dependencies right now, so jest will not run this test for current code actually.

Reply:
Oh, I got it.